composer/trainer/trainer.py

```diff
@@ -1612,14 +1720,27 @@ def fit(
 
             self.state.evaluators = evaluators
 
-        # Grad Accum
-        if grad_accum is not None:
-            self.state.auto_grad_accum = _is_auto_grad_accum(grad_accum, device=self._device)
-            if self.state.auto_grad_accum and self.state.profiler:
+        # Microbatching
+        if grad_accum is not None and train_device_microbatch_size is not None:
+            raise ValueError('Cannot specify both `grad_accum` and `train_device_microbatch_size`.')
+        elif train_device_microbatch_size is not None:
+            self.state.auto_microbatching = _is_auto_microbatching(train_device_microbatch_size, device=self._device)
+            if self.state.auto_microbatching and self.state.profiler:
+                raise ValueError("`train_device_microbatch_size='auto'` is not compatible with the profiler. It is "
+                                 "recommended to run a mini-run with `train_device_microbatch_size='auto'` to identify "
+                                 'the optimal train_device_microbatch_size value and then manually specify that in a '
+                                 'second run with profiler.')
+            self.state.train_device_microbatch_size = _get_initial_train_device_microbatch_size(
+                train_device_microbatch_size, self.state.train_dataloader)
+            self.state.using_device_microbatch_size = True
+        elif grad_accum is not None:
+            self.state.auto_microbatching = _is_auto_grad_accum(grad_accum, device=self._device)
+            if self.state.auto_microbatching and self.state.profiler:
                 raise ValueError("`grad_accum='auto'` is not compatible with the profiler. It is recommended to run "
                                  "a mini-run with `grad_accum='auto'` to identify the optimal grad_accum value and "
                                  'then manually specify that in a second run with profiler.')
             self.state.grad_accum = _get_initial_grad_accum(grad_accum)
+            self.state.using_device_microbatch_size = False
```

Comment:
aren't these two if blocks basically doing the same thing? can't you just map grad_accum to to microbatch_size and then do all the microbatch processing? instead of handling both these cases?
can't you just do:
```python
if grad_accum is not None:

  if grad_accum == 'auto':
    microbatch_size = 'auto'
  else:
    microbatch_size = batch_size // grad_accum

 do_microbatch_size_stuff()

```
or whatever the conversion is

Reply:
Same comment as before -- batch_size isn't known a priori

Reply:
so then when we deprecate grad_accum there is no option for not specifying how you want to scale your batch size on a device: you either specify microbatch_size = 'auto' or you specify a constant?

Reply:
Correct. Instead of specifying a scalar (grad_accum) you directly specify batch size (microbatch_size) which is more intuitive and easier to figure out

Reply:
ah ok makes sense. thanks!